### PR TITLE
fix(internal): Also stop process when machine is just created

### LINKED ISSUE #1726: ``kraft rm`` does not kill ``qemu`` process if the VM is not started
### Describe the bug

Run a kernel without starting it. Then remove it. The ``qemu`` process is never killed.

### Steps to reproduce

```bash
$ ps aux | grep qemu
305:kero       27961  0.0  0.0  12100  2688 pts/1    S+   16:21   0:00 grep -n --color qemu

$ kraft run nginx:1.25 --no-start
nervous_maggie

$ kraft ps --all
NAME            KERNEL            ARGS            CREATED        STATUS   MEM  PORTS  PLAT
nervous_maggie  oci://nginx:1.25  /usr/bin/nginx  4 seconds ago  created  64M         qemu/x86_64

$ kraft rm --all
nervous_maggie

$ kraft ps --all
NAME  KERNEL  ARGS  CREATED  STATUS  MEM  PORTS  PLAT

$ ps aux | grep qemu
304:root       27997  0.5  0.2 383056 70276 ?        Sl   16:21   0:00 qemu-system-x86_64 -append /usr/bin/nginx  -cpu host,+x2apic,-pmu -daemonize -device pvpanic -device sga -display none -enable-kvm -kernel /home/kero/.local/share/kraftkit/runtime/e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9/unikraft/bin/kernel -machine pc,accel=kvm -m size=64M -monitor unix:/home/kero/.local/share/kraftkit/runtime/e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9/qemu_mon.sock,server,nowait -name e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9 -nographic -no-reboot -S -parallel none -pidfile /home/kero/.local/share/kraftkit/runtime/e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9/machine.pid -qmp unix:/home/kero/.local/share/kraftkit/runtime/e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9/qemu_control.sock,server,nowait -qmp unix:/home/kero/.local/share/kraftkit/runtime/e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9/qemu_events.sock,server,nowait -rtc base=utc -serial file:/home/kero/.local/share/kraftkit/runtime/e9da9bb7-5e1c-415c-bd58-3fd4cd2fd3b9/machine.log -smp cpus=1,threads=1,sockets=1 -vga none
308:kero       28074  0.0  0.0  12100  2688 pts/1    S+   16:21   0:00 grep -n --color qemu
```

### Expected behavior

For the qemu process to be cleaned

### Which architectures were you using or does this bug affect?

_No response_

### Which operating system were you using or does this bug affect?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by craciunoiuc**
GitHub-Closes: #1726

### Prerequisite checklist

  - [ ] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [ ] Tested your changes against relevant architectures and platforms;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### CODE REVIEW

File: internal/cli/kraft/remove/remove.go [hunk 1]
```diff
@@ -229,7 +229,11 @@ func (opts *RemoveOptions) Run(ctx context.Context, args []string) error {
 		}
 
 		// Stop the machine before deleting it.
-		if machine.Status.State == machineapi.MachineStateRunning {
+		if machine.Status.State == machineapi.MachineStateRunning ||
+			machine.Status.State == machineapi.MachineStateCreated ||
+			machine.Status.State == machineapi.MachineStatePaused ||
+			machine.Status.State == machineapi.MachineStateRestarting ||
+			machine.Status.State == machineapi.MachineStateSuspended {
```
Comment (nderjung) [inline internal/cli/kraft/remove/remove.go]: Is the if-statement even necessary in fact? The error is just logged and not returned.

Comment (craciunoiuc) [reply internal/cli/kraft/remove/remove.go]: depends on how you like it, take the case of running `rm --all` after a long time

You will be spammed by 20-30 error messages, when in fact all is good cause you want those removed

Comment (LucaSeri) [reply internal/cli/kraft/remove/remove.go]: What about attempting to stop it and then logging the error in case the State is not Exited?

Comment (craciunoiuc) [reply internal/cli/kraft/remove/remove.go]: Sure, but that's more or less the same thing right? It's just more code being executed

I can flip the `if` and check if the machine is not Exited/Errored/Unknown/Failed, but that's only 2 conditions  less, not sure if worth the hassle 😅

Comment (nderjung) [reply internal/cli/kraft/remove/remove.go]: What happens if the instance is stopped and the `Stop` method is called again? I.e. does it error? If it does not error, we should just call this method regardless as it acts a safety mechanism for ensuring the machine is shutdown.

Comment (craciunoiuc) [reply internal/cli/kraft/remove/remove.go]: removed if case

**Review by nderjung — APPROVED** [review]
Thanks!

**Pull request merged**